atomix/cluster/src/main/java/io/atomix/raft/roles/LeaderAppender.java

```diff
@@ -556,9 +563,12 @@ private void completeCommits(final long previousCommitIndex, final long commitIn
     for (long i = previousCommitIndex + 1; i <= commitIndex; i++) {
       final CompletableFuture<Long> future = appendFutures.remove(i);
       if (future != null) {
+        metrics.observeCommit();
         future.complete(i);
       }
     }
+
+    metrics.observeNonCommittedEntries(raft.getLog().getLastIndex() - commitIndex);
```

Comment:
:x: This must be also updated on append. Otherwise, if nothing is committed we won't see it reflected in the metrics.